### PR TITLE
Syncing with uio GitHub

### PR DESCRIPTION
**Pull request by branislavjenco**
Hi! 
I noticed that the repository on UiO's github and here diverged. This PR is a collection of commits from the UiO repository that have been made roughly since when these two repos diverged. I extracted them as git patches from the UiO repo and applied them here. 

These include the changes made by Henry since 22. 5. 2020, as well as commits made by Carsten on 28. 8. 2020 (all on the UiO github). The last commit at the end is the one made by Carsten in this repo where he updated the readme, reapplied. 

There is one little divergence in the `registration_estimation.cpp/translation_estimation` function where in UiO's repo, `covariance_sampling` was used, but here, it was changed to `normal_space_sampling` in the commit `ae147168`. This PR keeps the one that was used here.

### CODE REVIEW

File: include/load_data.h [hunk 1]
```diff
@@ -1,32 +1,30 @@
...
 
 #include <pcl/point_cloud.h>
 
-
-#include <Eigen/dense>
- 
-
-#include "data_types.h"
-
+#if defined __GNUC__ || defined __APPLE__
```
Comment (griwodz) [inline include/load_data.h:16]: Wouldn't it be better to find the Eigen base directory in CMake and add eigen3 to the target_include_path when necessary?
I think it depends on the specific package manager if it ends up in the eigen3 subdir or not, it's not the OS's problem.

File: reconstruction++.cpp [hunk 2]
```diff
@@ -1,217 +1,107 @@
...
-
-       std::cout << "Fragment pairwise registration" << std::endl << std::endl; 
-
-
-       std::vector <pcl::PointCloud <pcl::PointXYZ> > fragment_clouds = load_fragments(data_dir + "/fragments"); 
-
-       incremental_pairwise_registration(fragment_clouds, translations, data_dir, visualization); 
-
-
+    if (boost::filesystem::exists(data_dir + "/odometry")) {
```
Comment (griwodz) [inline reconstruction++.cpp:88]: It would have been a bit easier to have the removal of empty lines in a separate PR. But probably too difficult now.

Comment (branislavjenco) [reply reconstruction++.cpp:88]: Actually, I think to split it up wouldn't be too hard. I'll try and make this into two PRs

Comment (branislavjenco) [reply reconstruction++.cpp:88]: I'm going to force push this PR without the reformatting commits and make another branch that only applies those

File: include/load_data.h (hunk 1 above)
Comment (branislavjenco) [reply include/load_data.h:16]: Indeed, that sounds like a good idea. I only focus on bringing the commits from the other repo here in this PR but this is something we should fix.

File: reconstruction++.cpp (hunk 2 above)
Comment (branislavjenco) [reply reconstruction++.cpp:88]: Done. This one is now without the reformatting. The build seems to work as well. After we merge this, I'll make a PR with the reformatting, and after that another one where I updated PCL to 1.11 and added the symmetric objective option for ICP.

**Review by henryhansen23 — APPROVED** [review]

**Pull request merged**